### PR TITLE
feat: add Integration Hub client onboarding

### PR DESCRIPTION
**Pull request by plotline-insights**
## Summary
- Adds the Integration Hub onboarding portfolio, client workspace, projects, work items, risks, blockers, timeline, and source links.
- Uses Loma's existing authenticated session and module permissions. No separate proxy secret or account-grant system is introduced.
- Adds strictly read-only Slack, Grain, and Pylon ingestion. These connectors cannot send, reply, update, or create records in external systems.

## Onboarding capabilities
- Server-paginated portfolio with child-backed effective health and urgency sorting
- Account detail, onboarding plan, milestones, tasks, risks, blockers, projects, playbooks, dependencies, archive/restore, audit history, and curated timeline
- My Action Center and calculated health indicators
- ETags, conditional mutations, structured errors, request IDs, transactional audit writes, and idempotent account creation

## Read-only communication ingestion
- Immutable external IDs, including Slack's original full-precision `ts`
- Separate immutable raw events, normalized interactions, conversations, and evidence-backed unreviewed findings
- Account-scoped interaction deduplication for shared channels and meetings
- Explicit source mappings, incremental checkpoints, durable queued jobs, distributed mapping locks, retries, schedules, and sync logs
- Grain recording metadata, participants, summaries, action items, and transcript excerpts
- Pylon issue context attached to imported messages
- Evidence-backed rule classification that identifies likely questions/issues and waiting state
- Imported findings remain unreviewed and never modify official onboarding status automatically

## Data and reliability
- Separate collections for accounts, projects, tasks, milestones, risks/blockers, sources, interactions, sync jobs, timeline, audit, idempotency, and rate limits
- Mongo-backed distributed request limiting with a higher cost for connector sync requests
- Full dependency-graph cycle validation
- Account GET returns the ETag from the same hydrated version as the response body
- Audit limits are clamped and account creation reuses a form-session idempotency key across retries

## Testing
- Integration Hub focused tests: 49 passed
- Dashboard targeted ESLint: passed
- Dashboard production build: passed
- Full backend suite: 330 passed; one unrelated pre-existing client-pool initialization test failed
- Python compilation and `git diff --check`: passed

## Safety
The ingestion path imports information only. It is isolated from Loma agent responders and exposes no Slack/Pylon/Grain send or mutation operation.

### CODE REVIEW

File: integration_hub/service.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        now = datetime.now(timezone.utc)
+        item = {
+            "item_id": f"item_{uuid.uuid4()}",
+            **normalize_work_item(data),
+            "created_at": now,
+            "created_by": actor,
+            "updated_at": now,
+            "updated_by": actor,
+        }
+        account = await self.repository.add_work_item(account_id, item)
```
Comment (plotline-insights) [inline integration_hub/service.py]: 🔴 **BLOCKING**: Work-item, activity, and source-link mutations never advance the account's `updated_at`, `updated_by`, or `version`.

As a result, the workspace's "Last updated" value and the repository's updated-at ordering remain stale after most onboarding activity. This also means `key={account.version}` will not reset the account form after these mutations. Please make every account mutation atomically touch the parent metadata (and ideally append its audit activity in the same update).

File: integration_hub/models.py [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+        raise ValidationError("stage is invalid")
+    if health not in HEALTH_STATES:
+        raise ValidationError("health is invalid")
+    return {
+        "name": _text(data.get("name"), "name", required=True, maximum=200),
+        "status": "active",
+        "owner_email": normalize_email(data.get("owner_email")),
+        "stage": stage,
+        "health": health,
+        "health_override_enabled": bool(data.get("health_override_enabled", False)),
```
Comment (plotline-insights) [inline integration_hub/models.py]: 🔴 **BLOCKING**: Validate this as a real boolean instead of coercing it. For example, JSON `{ "health_override_enabled": "false" }` currently enables the override because `bool("false")` is `True`.

The same coercion exists for work-item `escalated`. Please apply strict boolean validation on create and work-item create/update, matching `normalize_update`.

File: integration_hub/repository.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    def __init__(self, db):
+        self.collection = db.integration_accounts
+
+    async def create(self, account):
+        await self.collection.insert_one(account)
+        return account
+
+    async def list(self, query, limit=100):
+        return await self.collection.find(query).sort(
+            [("updated_at", -1), ("account_id", 1)]
```
Comment (plotline-insights) [inline integration_hub/repository.py:98]: 🔴 **BLOCKING**: This hard cap silently hides clients after the first 100. It also affects `list_actions`, so assigned tasks and at-risk accounts can disappear from My Action Center even though they remain incomplete.

Please add explicit pagination for the portfolio and ensure the action query scans all relevant records (preferably with a dedicated query/aggregation rather than relying on the portfolio limit).

File: dashboard/src/components/integration-hub/OnboardingWorkspace.tsx [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+          {TYPES.map((type) => {
+            const items = (account.work_items || []).filter((entry) => entry.type === type);
+            return (
+              <section key={type} className="rounded-lg border p-3">
+                <div className="mb-2 flex items-center justify-between"><h3 className="text-sm font-medium">{formatIntegrationLabel(type)}s</h3><Badge variant="secondary">{items.length}</Badge></div>
+                {items.length === 0 ? <p className="text-xs text-muted-foreground">No {type}s added.</p> : (
+                  <div className="space-y-2">
+                    {items.map((entry) => (
+                      <div key={entry.item_id} className="flex flex-wrap items-center gap-2 rounded-md bg-muted/40 p-2">
+                        <div className="min-w-48 flex-1"><p className="text-sm font-medium">{entry.title}</p><p className="text-xs text-muted-foreground">{entry.owner_email || "Unassigned"}{entry.due_at ? ` · Due ${new Date(entry.due_at).toLocaleDateString()}` : ""}{entry.severity ? ` · ${formatIntegrationLabel(entry.severity)}` : ""}</p></div>
```
Comment (plotline-insights) [inline dashboard/src/components/integration-hub/OnboardingWorkspace.tsx]: 🟡 **SUGGESTION**: The model/API support description, dependency, resolution, escalation, owner, due date, and severity updates, but the workspace only renders summary text and allows status changes or deletion. That prevents the dashboard from actually maintaining several of the details this feature is meant to centralize.

Consider an edit action/form for the full work item, and display the stored description/dependency/resolution so those fields are usable rather than write-only through the API.

**Review by plotline-insights — COMMENTED** [review]
## Summary

Strong foundation and a useful dashboard structure, especially the derived health explanations, action-center concept, validation boundaries, and audit/source-link model. There are several blocking persistence and completeness issues that can make the hub silently show stale or incomplete onboarding data. GitHub does not allow this account to request changes on its own PR, so this is submitted as a comment review.

## Review

### 🔴 Blocking Issues
- Parent account metadata is not updated for work-item, activity, or source-link mutations, so last-updated ordering and timestamps become stale.
- Boolean coercion accepts strings such as `"false"` as true for health overrides and escalations.
- The repository silently caps both the portfolio and My Action Center source data at 100 accounts without pagination.
- The required `public-content-scan` check is currently failing because `@plotline.so` markers were added to public-scanned files.

### 🟡 Suggestions
- Expose editing and display for the full work-item schema, including description, dependency, resolution, escalation, owner, due date, and severity.
- Add service/repository/route tests. The 17 new tests cover normalization and health calculation well, but do not exercise persistence, audit metadata, HTTP behavior, concurrency, or the 100-record boundary.
- Escape literal search text before using it as a Mongo regex, both to prevent malformed-pattern failures and to avoid unexpectedly expensive regex queries.
- Surface errors for inline status/delete/source-delete actions instead of allowing rejected promises to fail silently.

### 🎉 What's Good
- Clear separation between route, service, repository, and validation layers.
- Useful calculated-health reasons rather than only a status badge.
- Good model-level coverage for the primary validation cases.
- CI backend, dashboard, preview, and secret-scan checks pass, and the targeted model suite passes locally (`17 passed`).

File: integration_hub/models.py [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
```
Comment (plotline-insights) [inline integration_hub/models.py]: 🔴 **BLOCKING**: The strict-boolean fix is still incomplete. `normalize_create` continues to coerce strings, so `{ "health_override_enabled": "false" }` becomes `true`; `normalize_work_item` does the same for `escalated` on both create and update.

Please reject non-boolean values in both normalizers, consistent with `normalize_update`, and add regression tests for each path.

File: integration_hub/repository.py [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+                "$set": {"updated_at": project["created_at"], "updated_by": project["created_by"]},
+            },
+        )
+        return await self.get(account_id) if result.matched_count else None
+
+    async def append_activity(self, account_id, activity):
+        await self.collection.update_one(
+            {"account_id": account_id, "archived_at": None},
+            {"$push": {"activities": activity}},
+        )
```
Comment (plotline-insights) [inline integration_hub/repository.py:237]: 🔴 **BLOCKING**: Most child mutations still do not touch the parent account metadata. `append_activity`, source-link mutations, and work-item mutations leave `updated_at`, `updated_by`, and `version` unchanged; the service also writes the child change and audit entry in two separate operations.

This keeps portfolio ordering and “Last updated” stale and does not provide the atomic audit/version behavior described by the PR. Please make each mutation a single atomic update that changes the child data, appends the activity, updates the parent metadata, and increments/checks `version`.

File: integration_hub/service.py [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+    async def get(self, account_id, include_archived=False):
+        account = (
+            await self.repository.get_any(account_id)
+            if include_archived else await self.repository.get(account_id)
+        )
+        return self.enrich(account) if account else None
+
+    async def list_actions(self, actor):
+        now = datetime.now(timezone.utc)
+        accounts = await self.repository.list({"archived_at": None})
```
Comment (plotline-insights) [inline integration_hub/service.py]: 🔴 **BLOCKING**: My Action Center still reads through `repository.list()` with its default `limit=100`, so assigned tasks and at-risk clients after the first 100 accounts remain invisible.

Please use a dedicated unbounded/paginated query or aggregation for actions and attention accounts rather than the portfolio list default.

File: dashboard/src/app/integration-hub/page.tsx [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+  const [attentionAccounts, setAttentionAccounts] = useState<IntegrationAccount[]>([]);
+  const [sort, setSort] = useState("urgency");
+  const [status, setStatus] = useState("active");
+
+  const load = useCallback(async () => {
+    setLoading(true);
+    setError(null);
+    try {
+      const [data, actionData] = await Promise.all([
+        fetchIntegrationAccounts({ search, stage: stage === "all" ? undefined : stage, status }),
```
Comment (plotline-insights) [inline dashboard/src/app/integration-hub/page.tsx]: 🔴 **BLOCKING**: The API now returns paginated results, but the dashboard never passes `page`/`page_size`, stores `pagination`, or renders pagination controls. The portfolio therefore silently stops at the default first 50 clients.

Please wire the returned pagination metadata into the UI (or implement infinite loading) so every client remains reachable.

**Review by plotline-insights — COMMENTED** [review]
## Summary

Re-reviewed the latest head (`0e4fc52`). The new archive/restore lifecycle, project/playbook support, account-update concurrency guard, backend pagination contract, and expanded model tests are meaningful improvements. However, several previously reported blockers remain, and the new pagination contract is not connected to the dashboard.

## Review

### 🔴 Blocking Issues
- Parent metadata and optimistic concurrency are still missing from activity, source-link, and work-item mutations; audit writes are also still separate operations.
- Boolean strings are still coerced for account creation and work-item escalation.
- My Action Center still silently scans only the first 100 accounts.
- The dashboard only shows the first 50 portfolio accounts because it ignores the new pagination metadata.
- The required `public-content-scan` check is still failing on the latest head due to `@plotline.so` markers in dashboard, test, and seed files.

### 🟡 Remaining Suggestions
- Escape literal search text before constructing the Mongo regex.
- Add service, repository, and route tests for persistence, atomic audit/version updates, `409` behavior, archive/restore, and pagination boundaries.
- Add full work-item editing/display and visible error handling for inline status/delete actions.
- Reject unknown project playbook IDs instead of silently converting them to “No playbook.”

### 🎉 What's Good
- Account updates now use an atomic audit write and version check.
- Archive/restore transitions and project playbooks are useful additions.
- Model tests increased from 17 to 21 and pass locally.
- Local targeted tests, compilation, and `git diff --check` pass; GitHub backend, dashboard, preview, and secret-scan checks are green.

GitHub does not allow this account to request changes on its own PR, so this is submitted as a comment review.

File: api/integration_hub_routes.py [hunk 9]
```diff
@@ -0,0 +1,301 @@
...
+
+
+def _context(request):
+    db = get_db()
+    if db is None:
+        raise web.HTTPServiceUnavailable(
+            text='{"error":"Observability not configured"}',
+            content_type="application/json",
+        )
+    actor = get_user_email(request)
```
Comment (plotline-insights) [inline api/integration_hub_routes.py:55]: 🔴 **BLOCKING**: Authentication alone is not sufficient for this module. The Phase 0 contract requires a backend feature flag, explicit Integration Hub permissions, and account-scoped access grants on every request, with an empty allowlist denying access. As written, every authenticated Loma user can list, read, create, and mutate every customer's onboarding record. Please implement the authentication prerequisite/PR 0 authorization gate before exposing these routes, rather than treating authorization as a later phase.

File: integration_hub/repository.py [hunk 10]
```diff
@@ -0,0 +1,143 @@
+"""MongoDB persistence for Integration Hub accounts."""
+
+
+class AccountRepository:
+    def __init__(self, db):
+        self.collection = db.integration_accounts
```
Comment (plotline-insights) [inline integration_hub/repository.py:20]: 🔴 **BLOCKING**: This storage model does not implement the Phase 0 data contract. Projects, milestones, tasks, risks, source mappings, access grants, and audit logs are specified as separate `integration_*` collections with their own public IDs, versions, indexes, and lifecycle rules. Embedding all children and the "audit log" in one account document prevents independent optimistic concurrency and pagination, makes the document grow without bound toward MongoDB's document limit, and cannot provide the required append-only audit record with before/after state. Please preserve the specified collection boundaries and use a MongoDB transaction for each resource mutation plus audit/access-grant writes.

File: api/integration_hub_routes.py [hunk 11]
```diff
@@ -0,0 +1,301 @@
...
+    )
+    app.router.add_post(
+        "/api/integration-hub/accounts/{account_id}/work-items",
+        handle_create_work_item,
+    )
+    app.router.add_patch(
+        "/api/integration-hub/accounts/{account_id}/work-items/{item_id}",
+        handle_update_work_item,
+    )
+    app.router.add_delete(
```
Comment (plotline-insights) [inline api/integration_hub_routes.py]: 🔴 **BLOCKING**: The design explicitly says hard delete is not exposed and that lifecycle changes are audited and reversible. These DELETE routes physically remove work items/source links from the embedded record and lose the data needed for a full before/after audit. Replace them with the defined archive/cancel/transition semantics, including permission checks, `If-Match`, reason where required, and an append-only audit entry.

File: dashboard/src/lib/integration-hub-api.ts [hunk 12]
```diff
@@ -0,0 +1,253 @@
...
+  return request(`/api/integration-hub/accounts${suffix}`);
+}
+
+export async function fetchAllIntegrationAccounts(filters: {
+  search?: string; stage?: string; health?: string; owner?: string; status?: string;
+} = {}): Promise<IntegrationAccount[]> {
+  const pageSize = 100;
+  const accounts: IntegrationAccount[] = [];
+  let page = 1;
+  while (true) {
```
Comment (plotline-insights) [inline dashboard/src/lib/integration-hub-api.ts]: 🟡 **PERFORMANCE / CONTRACT**: Fetching every page sequentially on each search/filter change defeats pagination and makes initial render latency and memory usage grow linearly with the portfolio. It also uses offset pagination against a mutable `updated_at` sort, so concurrent changes can duplicate or skip accounts. Keep pagination server-driven in the UI (or use an opaque cursor/infinite list), and query Action Center directly with indexed aggregation instead of loading all accounts into Python.

File: integration_hub/models.py [hunk 13]
```diff
@@ -0,0 +1,342 @@
...
+        "dependency": _text(data.get("dependency"), "dependency", maximum=500),
+        "resolution": _text(data.get("resolution"), "resolution"),
+        "escalated": _boolean(data.get("escalated"), "escalated"),
+        "project_id": _text(data.get("project_id"), "project_id", maximum=100),
+        "depends_on": _text_list(data.get("depends_on"), "depends_on", maximum_items=20),
+    }
+
+
+def normalize_project(data):
+    status = data.get("status") or "planned"
```
Comment (plotline-insights) [inline integration_hub/models.py]: 🔴 **BLOCKING**: The implemented project/work-item lifecycle is incompatible with the approved contract. Projects should use lifecycle status `active|paused|completed|cancelled`, separate `stage`, explained health, archive metadata, and versioning; tasks, milestones, and risks each have distinct statuses and transition/reopen rules. Here `planned|blocked|archived` is used as project status, and all work-item types share `not_started|in_progress|blocked|completed`, which will require a data migration before later phases. Please align the schema and transition validators now, before pilot data is created.

**Review by plotline-insights — COMMENTED** [review]
## Summary

I compared the latest head (`e34327c`) against the restricted Phase 0 specification, not only the PR description. The latest commits correctly fix the earlier boolean coercion, stale parent metadata, first-100 Action Center cap, dashboard first-page cap, MongoDB naive datetime handling, and public-content scan. However, the PR is not safe to merge as Phase 1 because it skips the prerequisite and PR 0 gates and implements materially different persistence, authorization, lifecycle, and API contracts.

## 🔴 Blocking before merge

- Add the authentication prerequisite, backend feature flag, explicit permission checks, and account-scoped access grants. The current module grants every authenticated user full access to every account.
- Replace the single embedded account document with the specified `integration_*` resource collections and append-only audit collection. Resource plus audit/access-grant changes must commit in a MongoDB transaction.
- Align project, milestone, task, and risk schemas and lifecycle transitions with the approved contract before creating pilot data.
- Do not expose physical DELETE operations. Use the defined archive/cancel/transition and authorized reopen behavior.
- Implement the API contract: opaque cursor pagination, `ETag`/required `If-Match` with `412`, `Idempotency-Key`, rate limits, stable error shape/request IDs, filter validation, and HTTPS-only source URLs.
- Split this into the planned prerequisite, PR 0 validation slice, and reviewable Phase 1 PRs. The specification explicitly says full Phase 1 must not begin until deployed PR 0 evidence is reviewed.

## 🟡 Improvements before the next phase

- Keep portfolio pagination server-driven rather than downloading every account after each search; query Action Center through indexed aggregation.
- Add repository, service, route, authorization, transaction rollback, concurrency, cross-account isolation, and dashboard state tests. The current 25 tests are almost entirely normalization/health tests.
- Validate project IDs and dependencies, prevent cycles/dangling references, and reject unknown playbooks instead of silently converting them to no playbook.
- Escape literal MongoDB search input and enforce maximum search length.
- Add full edit/display support for descriptions, dependencies, resolutions, escalation, ownership, dates, and severity, with visible error/conflict handling.

## ✅ Verified improvements

- Previous parent metadata/version, boolean coercion, pagination-cap, datetime, and public scan blockers are fixed.
- Latest CI is fully green.
- Local checks pass: `25 passed`, Python compilation, and `git diff --check`.

GitHub does not allow this account to request changes on its own PR, so this is submitted as a comment review with blocking findings.

File: integration_hub/models.py [hunk 14]
```diff
@@ -0,0 +1,351 @@
...
+
+
+def calculate_account_health(account, now=None):
+    """Return the derived health, explanations, and urgency counters."""
+    now = as_utc(now or datetime.now(timezone.utc))
+    upcoming_cutoff = now + timedelta(days=7)
+    open_items = [
+        item for item in account.get("work_items", [])
+        if item.get("status") != "completed"
+    ]
```
Comment (plotline-insights) [inline integration_hub/models.py:497]: 🔴 **BLOCKING**: Terminal states other than `completed` are still treated as open. A resolved blocker, accepted/resolved risk, achieved/cancelled milestone, or cancelled task can therefore remain overdue and keep the account `blocked` or `at_risk` forever.

Use terminal statuses per resource type, for example task `completed|cancelled`, milestone `achieved|cancelled`, and risk/blocker `resolved|accepted` where applicable. Please add regression tests showing a resolved blocker and resolved risk no longer affect health.

File: integration_hub/repository.py [hunk 15]
```diff
@@ -0,0 +1,250 @@
...
+            {"$unwind": "$account"},
+            {"$match": {"account.archived_at": None}},
+            {"$sort": {"due_at": 1, "resource_id": 1}}, {"$limit": 500},
+        ]
+        actions = []
+        for collection in (self.tasks, self.milestones):
+            actions.extend(await collection.aggregate(pipeline).to_list(500))
+        attention = await self.collection.find({
+            "account_id": {"$in": ids}, "archived_at": None,
+            "$or": [{"health": {"$in": ["blocked", "at_risk", "escalated"]}}, {"current_blocker": {"$ne": None}}],
```
Comment (plotline-insights) [inline integration_hub/repository.py]: 🔴 **BLOCKING**: Action Center checks the stored `account.health`, while the portfolio displays `effective_health` calculated at read time from overdue work, blockers, and severe risks. This means an account can visibly be `blocked`/`at_risk` in the portfolio but never appear under “clients that need attention.”

Derive attention using the same source of truth as `calculate_account_health`, or persist/recompute the calculated health consistently. Add a test for an `on_track` account with an open blocker or overdue task appearing in attention.

File: api/integration_hub_routes.py [hunk 16]
```diff
@@ -0,0 +1,312 @@
...
+
+def _module_role(request):
+    if get_system_role(request) == "admin": return "Admin"
+    return (((request.get("user") or {}).get("tool_assignments") or {}).get("integration_hub") or {}).get("role")
+
+
+def _require_module(request, write=False):
+    role = _module_role(request)
+    allowed = {"Admin", "Analyst"} if write else {"Admin", "Analyst", "Read-only", "Support"}
+    if role not in allowed: raise web.HTTPForbidden(text="Integration Hub permission required")
```
Comment (plotline-insights) [inline api/integration_hub_routes.py:78]: 🔴 **BLOCKING / SIMPLIFY**: This adds a second onboarding-specific authorization layer that is not required by Loma’s product architecture. Integration Hub is part of the authenticated internal Loma dashboard, so users should reuse the existing Loma identity and intended internal role/tool visibility rather than requiring separate per-account grants before any data appears.

As implemented, existing authenticated users receive `403` unless someone first provisions a new `integration_hub` tool assignment, and non-admin users only see accounts for which a separate grant was created. The newly documented `LOMA_PROXY_SECRET` is also not consumed anywhere. Please simplify this around the existing Loma auth contract and remove unused deployment configuration. Keep account scoping only if there is a confirmed product requirement for private onboarding records.

File: integration_hub/models.py [hunk 17]
```diff
@@ -0,0 +1,351 @@
...
+    return {
+        "name": _text(data.get("name"), "name", required=True, maximum=200),
+        "description": _text(data.get("description"), "description"),
+        "status": status,
+        "stage": stage,
+        "health": health,
+        "health_reason": _text(data.get("health_reason"), "health_reason"),
+        "owner_email": normalize_email(data.get("owner_email")),
+        "target_at": normalize_date(data.get("target_at")),
+        "playbook": data.get("playbook") if data.get("playbook") in PLAYBOOKS else None,
```
Comment (plotline-insights) [inline integration_hub/models.py]: 🟡 **SUGGESTION**: Reject a supplied unknown playbook instead of silently converting it to `None`. A typo from the dashboard or API would create a project without its expected onboarding checklist, with no error explaining why.

**Review by plotline-insights — COMMENTED** [review]
## Summary

I re-reviewed the latest head (`96cb707f`) using the planning document as guidance, not as an unquestionable contract. I also corrected my earlier assumption: Loma already authenticates internal dashboard users, so a new authentication prerequisite is not required. The latest changes resolve the prior persistence, pagination, lifecycle, concurrency, and audit concerns, but they also introduce authorization complexity that does not match the clarified product intent.

## 🔴 Blocking before merge

- Simplify Integration Hub access around Loma’s existing authentication and intended internal permissions. Separate account grants and the unused `LOMA_PROXY_SECRET` should not be required unless private account-level visibility is a confirmed product requirement.
- Fix health calculation so terminal states such as resolved blockers/risks, achieved milestones, and cancelled items are not treated as open forever.
- Make Action Center use the same effective/calculated health semantics as the portfolio. Today an account can display as blocked or at risk but be absent from “needs attention.”

## 🟡 Improvements

- Reject unknown playbook IDs rather than silently dropping them.
- Escape and bound literal search input before using it as a MongoDB regex.
- Add route/repository integration tests for existing Loma auth reuse, calculated Action Center health, transaction rollback, cursor boundaries, and concurrency conflicts. The current 31 tests are useful but primarily model/contract tests.
- Consider replacing prompt-based work-item editing with a small structured form before wider internal adoption.

## ✅ Verified

- Separate bounded collections and indexes are now implemented.
- Child writes update the parent version and audit record transactionally.
- Strict boolean validation, cursor pagination, archive flows, visible mutation errors, and richer work-item editing are present.
- Targeted tests pass locally: `31 passed`.
- All current GitHub checks are green.

GitHub does not allow this account to request changes on its own PR, so this is submitted as a comment review with blocking labels.

File: api/integration_hub_routes.py [hunk 18]
```diff
@@ -0,0 +1,297 @@
...
+async def handle_create_account(request):
+    async def action():
+        service, actor, _ = _context(request); _require_module(request, True)
+        key = request.headers.get("Idempotency-Key", "").strip()
+        if not key: raise ValidationError("Idempotency-Key header is required")
+        cached = await service.repository.find_idempotent(actor, key)
+        if cached: return _ok(request, cached["response"], 200, cached["response"]["account"]["version"])
+        account = await service.create(await _json(request), actor, request["request_id"])
+        payload = {"account": _serialize(account)}
+        await service.repository.save_idempotent(actor, key, payload)
```
Comment (plotline-insights) [inline api/integration_hub_routes.py]: 🔴 **BLOCKING**: Account creation is not atomic with the idempotency record. Two concurrent requests with the same `Idempotency-Key` can both miss line 108, create different accounts, and then race on `save_idempotent`; each caller may receive a different account even though the key is identical.

Reserve the `(actor, key)` record before creating, or create the account and idempotency response in the same MongoDB transaction with a unique key. On a duplicate-key race, return the already stored response. Please add a concurrent-request regression test.

File: integration_hub/repository.py [hunk 19]
```diff
@@ -0,0 +1,225 @@
...
+                    ("integration_risks", "health_risks"),
+                )
+            ],
+            {"$addFields": {"work_items": {
+                "$concatArrays": ["$health_tasks", "$health_milestones", "$health_risks"]
+            }}},
+            {"$project": {"health_tasks": 0, "health_milestones": 0, "health_risks": 0}},
+            {"$sort": {"updated_at": -1, "account_id": 1}},
+        ]
+        attention = await self.collection.aggregate(attention_pipeline).to_list(None)
```
Comment (plotline-insights) [inline integration_hub/repository.py]: 🟡 **PERFORMANCE**: This fixes health consistency, but now every Action Center request joins all active accounts to all active tasks, milestones, and risks, materializes the full portfolio with `to_list(None)`, and only filters health afterward in Python. Runtime and memory will grow with all clients and historical work.

Push terminal-status, blocker, severity, due-date, and target-date predicates into the aggregation, return only accounts that can need attention, and apply a documented limit/cursor. That keeps the calculated-health behavior without hydrating the entire onboarding dataset.

**Review by plotline-insights — COMMENTED** [review]
## Summary

The latest commit addresses the previously reported functional issues: terminal lifecycle states no longer affect health, Action Center now uses calculated health, invalid playbooks are rejected, search expressions are escaped and bounded, per-account grants and the unused proxy secret were removed, and work-item editing now uses a structured form.

## Review

### 🔴 Blocking issue
- Account creation still has a concurrency hole in its idempotency implementation. The account and idempotency record are written separately, so simultaneous requests with the same key can create and return different accounts.

### 🟡 Optimization
- Action Center health is now correct, but its aggregation hydrates every active account and all active work items into memory before filtering. Push health predicates into MongoDB and bound/page the result before the portfolio grows.

### 🎉 What's good
- Existing Loma identity and module permissions are reused without adding a second account-grant authorization system.
- Health terminal-state handling now matches each work-item type.
- The structured edit form is substantially better than prompt-based editing.
- The 33 targeted tests pass locally, compilation and `git diff --check` pass, and all seven GitHub checks are green.

Because the connected GitHub account authored this PR, I submitted this as a comment review rather than `REQUEST_CHANGES`.
